src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java

```diff
@@ -95,26 +99,120 @@ private void setSubmissionData(Map<String, Object> data, String overseasEntityId
+        data.put(BENEFICIAL_OWNERS_INDIVIDUAL_FIELD, getBeneficialOwnersIndividualSubmissionData(submissionDto));
         data.put(BENEFICIAL_OWNERS_GOVERNMENT_OR_PUBLIC_AUTHORITY_FIELD, submissionDto.getBeneficialOwnersGovernmentOrPublicAuthority());
-        data.put(BENEFICIAL_OWNERS_CORPORATE_FIELD, submissionDto.getBeneficialOwnersCorporate());
+        data.put(BENEFICIAL_OWNERS_CORPORATE_FIELD, getBeneficialOwnersCorporateSubmissionData(submissionDto));
         data.put(MANAGING_OFFICERS_INDIVIDUAL_FIELD, submissionDto.getManagingOfficersIndividual());
         data.put(MANAGING_OFFICERS_CORPORATE_FIELD, submissionDto.getManagingOfficersCorporate());
         data.put(BENEFICIAL_OWNERS_STATEMENT, submissionDto.getBeneficialOwnersStatement());
 
-        String trustData = "";
-        if (Objects.nonNull(submissionDto.getTrusts())) {
-            // Convert trust data to JSON string if it exists on transaction else it's to an empty string
-            ObjectMapper mapper = JsonMapper.builder().findAndAddModules().build();
-            try {
-                trustData = mapper.writeValueAsString(submissionDto.getTrusts());
-            } catch (JsonProcessingException e) {
-                throw new ServiceException("Error converting trust data to JSON " + e.getMessage(), e);
+        ApiLogger.debug("Submission data has been set on filing", logMap);
+    }
+
+    private List<BeneficialOwnerIndividualDto> getBeneficialOwnersIndividualSubmissionData(OverseasEntitySubmissionDto submissionDto) throws ServiceException {
+        List<BeneficialOwnerIndividualDto> beneficialOwnersIndividualSubmissionData = new ArrayList<>();
+
+        if (Objects.nonNull(submissionDto.getBeneficialOwnersIndividual())) {
+            for (BeneficialOwnerIndividualDto beneficialOwner : submissionDto.getBeneficialOwnersIndividual()) {
+                List<TrustDataDto> trustsDataForBO = new ArrayList<>();
+                // Loop through each trustId(s) and lookup the information for that trust
+                if (Objects.nonNull(beneficialOwner.getTrustIds())) {
+                    for (String trustId : beneficialOwner.getTrustIds()) {
+                        List<TrustDataDto> trustData = new ArrayList<>();
+
+                        if (!Objects.nonNull(submissionDto.getTrusts())) {
+                            throw new ServiceException("No trusts exist for this filing but a trust id is provided for BO Individual "
+                                    + beneficialOwner.getFirstName() + " " + beneficialOwner.getLastName());
+                        }
+                        for (TrustDataDto trust : submissionDto.getTrusts()) {
+                            if (trust.getTrustId().equals(trustId)) {
+                                trustData.add(trust);
+                            }
+                        }
+
+                        // If there is more than one trust with the same ID throw an error
```

Comment:
I think this check is ensuring that a beneficial owner can only be linked to one trust? If yes, suggest rephrasing the comment and exception message just to make this a little clearer.

I might be wrong though, because if that's the case, why do we support multiple trust ids per BO?